### ISSUE TITLE
Restic mover job should fail if restic repo is locked

### ISSUE THREAD

**Issue by onedr0p** [backube/volsync#1429]
> @tesshuflower I wonder if we could make use of the [pod failure policy](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-failure-policy) feature and [restics exit codes](https://restic.readthedocs.io/en/latest/075_scripting.html#exit-codes) for a locked repo?
> 
> e.g. set this on the job...
> 
> ```yaml
> podFailurePolicy:
>   rules:
>     - action: FailJob
>       onExitCodes:
>         containerName: restic
>         operator: In
>         values: [11]
> ```
> 
> That should mark the job as failed and not be retried until the next volsync schedule instead of same job retrying over and over on a locked restic repo. 

 _Originally posted by @onedr0p in [#1415](https://github.com/backube/volsync/issues/1415#issuecomment-2429417481)_

**Comment by PrivatePuffin** [comment]
I want to add to this that this issue is even worse.
Got into an issue where a mover repeatedly fetched 99Gb(!) from S3 storage by running over-and-over trying to put 120Gb into a 100Gb PVC.

The sizing issue was my mistake, but it shouldn't try to repeatedly restart and burn through my S3 funds either.

**Comment by PrivatePuffin** [comment]
Yeah maybe 1 retry by default is excusable, but certainly with the S3 backend users can get into BOATLOADS of issues financially if things go horribly wrong.

**Comment by tesshuflower** [comment]
Currently the design of volsync is essentially very kubernetes centric, preferring to retry until things work.

If we just fail the job then we have no chance to retry for even something like a network hiccup.  If we could detect specific errors from restic, this is an interesting idea however.  If we did something it would definitely need to be an opt-in feature.  

One issue is that simply stopping the job from re-trying will not solve the scheduling issue, as volsync will not schedule until the previous synchronization has completed (i.e. the job has completed). VolSync will still reschedule the job after the backupLimit is hit. 
 Would need modifications to the scheduling code to mark the synchronization as completed with error.  Then there's still the question of whether we hang forever or retry on a schedule (which could still get someone into the same situation).

**Comment by PrivatePuffin** [comment]
@tesshuflower I agree, a few retries is not a bad thing.
But having it endlessly retry is going to cause more trouble than its solving things.

**Comment by samip5** [comment]
Yeah, my S3 bill is the way I noticed the whole problem after the fact..

**Comment by onedr0p** [comment]
>  If we could detect specific errors from restic, this is an interesting idea however. If we did something it would definitely need to be an opt-in feature.

That's good to hear and I am glad you agree. There's no point in retrying if we know from restic exit codes if the job will never succeed. Hopefully there can be some improvements in this area.

**Comment by PrivatePuffin** [comment]
> Yeah, my S3 bill is the way I noticed the whole problem after the fact..

Same here, Luckily for me it wasn't the bill but it was the quotum warning

**Comment by onedr0p** [comment]
I am closing this issue as I moved onto another tool suited more to my needs: https://github.com/home-operations/kopiur/

Thanks to all the volsync contributors during my time using it, and if you feel like this is something you still want to tackle please don't hesitate to re-open.
